### PR TITLE
Nugg.ad:

### PR DESCRIPTION
**Pull request by numismatika**
- re-enable the ruleset

### CODE REVIEW

File: src/chrome/content/rules/Nugg.ad.xml [hunk 1]
```diff
@@ -1,22 +1,89 @@
...
+	<target host="icinga.nugg.ad" />
 
-	<target host="*.nugg.ad" />
 	<target host="*.nuggad.net" />
 
 	<securecookie host="^\.nuggad\.net$" name="^d$" />
 
 	<rule from="^http:"
 	        to="https:" />
+		<exclusion pattern="^http://mtm\.nuggad\.net/$" />
```
Comment (J0WI) [inline src/chrome/content/rules/Nugg.ad.xml]: Please add a comment why this exclusion is required and also a test url where the rule can be applied.

Comment (numismatika) [reply src/chrome/content/rules/Nugg.ad.xml]: it causes a redirect loop from 443->80. It is noted now.
Test url is already in 	<test url="http://mtm.nuggad.net/" />

Comment (J0WI) [reply src/chrome/content/rules/Nugg.ad.xml]: This test url matches the exclusion. There is no test url that works over https.

Comment (numismatika) [reply src/chrome/content/rules/Nugg.ad.xml]: which is intended as i haven't found any

Comment (J0WI) [reply src/chrome/content/rules/Nugg.ad.xml]: `http://mtm.nuggad.net/robots.txt`

File: src/chrome/content/rules/Nugg.ad.xml [hunk 2]
```diff
@@ -1,29 +1,99 @@
...
-	<test url="http://71.nuggad.net/" />
+	<test url="http://71i.nuggad.net/" />
 	<test url="http://71iv.nuggad.net/" />
 	<test url="http://asv.nuggad.net/" />
 	<test url="http://bei.nuggad.net/" />
 	<test url="http://ci.nuggad.net/" />
 	<test url="http://gwp.nuggad.net/" />
 	<test url="http://ip-s.nuggad.net/" />
 	<test url="http://mtm.nuggad.net/" />
+	<test url="http://mtm.nuggad.net/robots.txt/" />
```
Comment (J0WI) [inline src/chrome/content/rules/Nugg.ad.xml]: Trailing slash should be removed here.
Please have a look at the other issues exposed by Travis.

**Comment by numismatika** [comment]
@J0WI 
any idea why this cert fails?

`ERROR src/chrome/content/rules/Nugg.ad.xml: Fetch error: http://bei.nuggad.net/ => https://bei.nuggad.net/: (60, 'SSL certificate problem: unable to get local issuer certificate')`

According to SSLLabs and Hardenize we get the full 3 certificate long cert chain

**Comment by J0WI** [comment]
WFM too. @Hainish do we need to update root CAs for our tests?

**Comment by Folant** [comment]
@J0WI @numismatika 
Fails in Tor 7.5 however
kkkk

**Comment by J0WI** [comment]
Same with a new Firefox profile, but curl works. However, our tests are correct.

**Comment by Hainish** [comment]
@J0WI I'm getting intermittent failures.  One moment I see

```shell
user@https-everywhere ~/workspace/https-everywhere/.venv (cschanaj-fix-1097) $ curl https://bei.nuggad.net/                                           
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```

and the next I see

```shell
user@https-everywhere ~/workspace/https-everywhere/.venv (cschanaj-fix-1097) $ curl https://bei.nuggad.net/
<!DOCTYPE html>
<html>
  <body>
    <script type="text/javascript">
      document.write('<scr' + 'ipt type="text/javascript" src="//' + document.location.hostname + '/javascripts/nuggad-ls-peer.js"></scr' + 'ipt>');
    </script>
    <script type="text/javascript">
      nuggad.xpc.peer.init();
    </script>
  </body>
</html>
```

So I'd chock this up to the server doing something funky with certificates or load-balancing returning different servers with different certificates.  Indeed, in a DNS lookup we see a round-robin response:

```user@https-everywhere ~/workspace/https-everywhere/.venv (cschanaj-fix-1097) $ dig +short bei.nuggad.net
pbtngx-u-b-ds.nuggad.net.
pbtngx-u-ds.nuggad.net.
94.198.63.251
```

**Comment by Hainish** [comment]
I've confirmed the differential certificate delivery for the two round-robin responses I've encountered: `94.198.63.251` and `80.82.201.80`.

```shell
$ echo 1 | openssl s_client -connect 80.82.201.80:443
CONNECTED(00000003)
depth=3 C = SE, O = AddTrust AB, OU = AddTrust External TTP Network, CN = AddTrust External CA Root
verify return:1
depth=2 C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
verify return:1
depth=1 C = FR, ST = Paris, L = Paris, O = Gandi, CN = Gandi Standard SSL CA 2
verify return:1
depth=0 OU = Domain Control Validated, OU = Gandi Standard Wildcard SSL, CN = *.nuggad.net
verify return:1
---
Certificate chain
 0 s:/OU=Domain Control Validated/OU=Gandi Standard Wildcard SSL/CN=*.nuggad.net
   i:/C=FR/ST=Paris/L=Paris/O=Gandi/CN=Gandi Standard SSL CA 2
 1 s:/C=FR/ST=Paris/L=Paris/O=Gandi/CN=Gandi Standard SSL CA 2
   i:/C=US/ST=New Jersey/L=Jersey City/O=The USERTRUST Network/CN=USERTrust RSA Certification Authority
 2 s:/C=US/ST=New Jersey/L=Jersey City/O=The USERTRUST Network/CN=USERTrust RSA Certification Authority
   i:/C=SE/O=AddTrust AB/OU=AddTrust External TTP Network/CN=AddTrust External CA Root
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIE9zCCA9+gAwIBAgIRAPHnzwcjrX/jRjDPGc8+XG8wDQYJKoZIhvcNAQELBQAw
XzELMAkGA1UEBhMCRlIxDjAMBgNVBAgTBVBhcmlzMQ4wDAYDVQQHEwVQYXJpczEO
MAwGA1UEChMFR2FuZGkxIDAeBgNVBAMTF0dhbmRpIFN0YW5kYXJkIFNTTCBDQSAy
MB4XDTE4MDExMjAwMDAwMFoXDTIwMDExNjIzNTk1OVowYDEhMB8GA1UECxMYRG9t
YWluIENvbnRyb2wgVmFsaWRhdGVkMSQwIgYDVQQLExtHYW5kaSBTdGFuZGFyZCBX
aWxkY2FyZCBTU0wxFTATBgNVBAMMDCoubnVnZ2FkLm5ldDCCASIwDQYJKoZIhvcN
AQEBBQADggEPADCCAQoCggEBAOBc7DJtHRjPHWkzuuCnXRkUXXqCyfCzyMl4y6Om
sCJrVJwRz5ed3w5GlmWcndUulcdJT+TpsMkaPVM0ZBg4iAzOMO5Ytf6joBA0tEAY
EwN5FODPWjcQZ0VpYT4SVMqbZUvI1mONeQqRaTqk5qJm13EJTTSgba7n3xXwJykt
Z8XS4SzK6fX1A58YTPU0Yj3Z5ptujs67mhahGWXkyQy0nV7A0m39RjKwpc0MPo53
CBWjRgvxybLJhKWTGCm5i109HZclGyU4+h/kJ2yTtlI86wpzZIkAomYbLiantSAK
1HCABQuo3pRKAamQnutFQQU8efDRAe3O3+slkKmusLVohpkCAwEAAaOCAaswggGn
MB8GA1UdIwQYMBaAFLOQp9jJr07NYTyffK1df0H9aTDqMB0GA1UdDgQWBBSsQq9Y
b9Ot3ZQR5nf7imQY0vr7eTAOBgNVHQ8BAf8EBAMCBaAwDAYDVR0TAQH/BAIwADAd
BgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwSwYDVR0gBEQwQjA2BgsrBgEE
AbIxAQICGjAnMCUGCCsGAQUFBwIBFhlodHRwczovL2Nwcy51c2VydHJ1c3QuY29t
MAgGBmeBDAECATBBBgNVHR8EOjA4MDagNKAyhjBodHRwOi8vY3JsLnVzZXJ0cnVz
dC5jb20vR2FuZGlTdGFuZGFyZFNTTENBMi5jcmwwcwYIKwYBBQUHAQEEZzBlMDwG
CCsGAQUFBzAChjBodHRwOi8vY3J0LnVzZXJ0cnVzdC5jb20vR2FuZGlTdGFuZGFy
ZFNTTENBMi5jcnQwJQYIKwYBBQUHMAGGGWh0dHA6Ly9vY3NwLnVzZXJ0cnVzdC5j
b20wIwYDVR0RBBwwGoIMKi5udWdnYWQubmV0ggpudWdnYWQubmV0MA0GCSqGSIb3
DQEBCwUAA4IBAQAcTDAR1MTdwZfmvPsjqUrxithuI6nZGnB9IDHTYdIoDNdqYWQC
0j14mJ6esFf6P9F1Q3d8CRkB407lJuqAypR1+UY2ZVqqynr5WtxPjL0vyqdBkD5r
dUuE0cRj/CqPZXLMMCmfwnNUpCOP/3QvsAYLd6Oq8AxtZKvH0Ux9EzvhvcI2wklc
Icy2H6Kg24f2ywdjfXRtgZIx6wJqGZV0cSGm4Ey/wXGz+ZvU+xlqLnfD24MlTmYA
XDbJgzio1gDbCIfrfCrzzBg/ak90YiIj2akVWwtUnFfyb8HVeLqwy/QQmC2znw6V
7u78PLrYPT2dDCz/daBPpzDMryEPzz8iSbj3
-----END CERTIFICATE-----
subject=/OU=Domain Control Validated/OU=Gandi Standard Wildcard SSL/CN=*.nuggad.net
issuer=/C=FR/ST=Paris/L=Paris/O=Gandi/CN=Gandi Standard SSL CA 2
---
No client certificate CA names sent
---
SSL handshake has read 4358 bytes and written 603 bytes
---
New, TLSv1/SSLv3, Cipher is RC4-SHA
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : RC4-SHA
    Session-ID: 02C57213282E792788A28F80A53CC771F8FB76CBE7D3B82E054EFE1EAA239824
    Session-ID-ctx:
    Master-Key: 215827ECBA6F6C85EFFD39CD6026153278F84FD29607DEB196132E03F57CB607FC7E499CA1B792FC6FDCB28C7EE3103A
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1517503401
    Timeout   : 300 (sec)
    Verify return code: 0 (ok)
---
DONE
```

and

```shell
$ echo 1 | openssl s_client -connect 94.198.63.251:443
CONNECTED(00000003)
depth=0 OU = Domain Control Validated, OU = Gandi Standard Wildcard SSL, CN = *.nuggad.net
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 OU = Domain Control Validated, OU = Gandi Standard Wildcard SSL, CN = *.nuggad.net
verify error:num=27:certificate not trusted
verify return:1
depth=0 OU = Domain Control Validated, OU = Gandi Standard Wildcard SSL, CN = *.nuggad.net
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/OU=Domain Control Validated/OU=Gandi Standard Wildcard SSL/CN=*.nuggad.net
   i:/C=FR/ST=Paris/L=Paris/O=Gandi/CN=Gandi Standard SSL CA 2
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIE9zCCA9+gAwIBAgIRAPHnzwcjrX/jRjDPGc8+XG8wDQYJKoZIhvcNAQELBQAw
XzELMAkGA1UEBhMCRlIxDjAMBgNVBAgTBVBhcmlzMQ4wDAYDVQQHEwVQYXJpczEO
MAwGA1UEChMFR2FuZGkxIDAeBgNVBAMTF0dhbmRpIFN0YW5kYXJkIFNTTCBDQSAy
MB4XDTE4MDExMjAwMDAwMFoXDTIwMDExNjIzNTk1OVowYDEhMB8GA1UECxMYRG9t
YWluIENvbnRyb2wgVmFsaWRhdGVkMSQwIgYDVQQLExtHYW5kaSBTdGFuZGFyZCBX
aWxkY2FyZCBTU0wxFTATBgNVBAMMDCoubnVnZ2FkLm5ldDCCASIwDQYJKoZIhvcN
AQEBBQADggEPADCCAQoCggEBAOBc7DJtHRjPHWkzuuCnXRkUXXqCyfCzyMl4y6Om
sCJrVJwRz5ed3w5GlmWcndUulcdJT+TpsMkaPVM0ZBg4iAzOMO5Ytf6joBA0tEAY
EwN5FODPWjcQZ0VpYT4SVMqbZUvI1mONeQqRaTqk5qJm13EJTTSgba7n3xXwJykt
Z8XS4SzK6fX1A58YTPU0Yj3Z5ptujs67mhahGWXkyQy0nV7A0m39RjKwpc0MPo53
CBWjRgvxybLJhKWTGCm5i109HZclGyU4+h/kJ2yTtlI86wpzZIkAomYbLiantSAK
1HCABQuo3pRKAamQnutFQQU8efDRAe3O3+slkKmusLVohpkCAwEAAaOCAaswggGn
MB8GA1UdIwQYMBaAFLOQp9jJr07NYTyffK1df0H9aTDqMB0GA1UdDgQWBBSsQq9Y
b9Ot3ZQR5nf7imQY0vr7eTAOBgNVHQ8BAf8EBAMCBaAwDAYDVR0TAQH/BAIwADAd
BgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwSwYDVR0gBEQwQjA2BgsrBgEE
AbIxAQICGjAnMCUGCCsGAQUFBwIBFhlodHRwczovL2Nwcy51c2VydHJ1c3QuY29t
MAgGBmeBDAECATBBBgNVHR8EOjA4MDagNKAyhjBodHRwOi8vY3JsLnVzZXJ0cnVz
dC5jb20vR2FuZGlTdGFuZGFyZFNTTENBMi5jcmwwcwYIKwYBBQUHAQEEZzBlMDwG
CCsGAQUFBzAChjBodHRwOi8vY3J0LnVzZXJ0cnVzdC5jb20vR2FuZGlTdGFuZGFy
ZFNTTENBMi5jcnQwJQYIKwYBBQUHMAGGGWh0dHA6Ly9vY3NwLnVzZXJ0cnVzdC5j
b20wIwYDVR0RBBwwGoIMKi5udWdnYWQubmV0ggpudWdnYWQubmV0MA0GCSqGSIb3
DQEBCwUAA4IBAQAcTDAR1MTdwZfmvPsjqUrxithuI6nZGnB9IDHTYdIoDNdqYWQC
0j14mJ6esFf6P9F1Q3d8CRkB407lJuqAypR1+UY2ZVqqynr5WtxPjL0vyqdBkD5r
dUuE0cRj/CqPZXLMMCmfwnNUpCOP/3QvsAYLd6Oq8AxtZKvH0Ux9EzvhvcI2wklc
Icy2H6Kg24f2ywdjfXRtgZIx6wJqGZV0cSGm4Ey/wXGz+ZvU+xlqLnfD24MlTmYA
XDbJgzio1gDbCIfrfCrzzBg/ak90YiIj2akVWwtUnFfyb8HVeLqwy/QQmC2znw6V
7u78PLrYPT2dDCz/daBPpzDMryEPzz8iSbj3
-----END CERTIFICATE-----
subject=/OU=Domain Control Validated/OU=Gandi Standard Wildcard SSL/CN=*.nuggad.net
issuer=/C=FR/ST=Paris/L=Paris/O=Gandi/CN=Gandi Standard SSL CA 2
---
No client certificate CA names sent
---
SSL handshake has read 1934 bytes and written 415 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES256-GCM-SHA384
    Session-ID: 8DC2161905845212F42608BD027732B39F3938048EF4786CACABF784F1A3965A
    Session-ID-ctx:
    Master-Key: 6C992096E6139D114F010A4F6D1B7D1CBB236C756CA2E0A68427C3E888096069E7CF7BD65D7CA9118462F725A2F43089
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 300 (seconds)
    TLS session ticket:
    0000 - ad d6 44 b5 14 48 0b 57-85 ed 8d 4e 8d 8a fa af   ..D..H.W...N....
    0010 - 9c 55 93 ef 71 ab ec 78-3f 69 d0 0e 78 43 c8 21   .U..q..x?i..xC.!
    0020 - 76 fe ae 08 2a 9f 4d c4-eb 83 a5 51 8e 44 6e af   v...*.M....Q.Dn.
    0030 - 96 19 d8 e8 8f 9b 4d be-03 53 7b 2d f2 d2 c0 43   ......M..S{-...C
    0040 - 02 2f f5 25 8f 71 62 9f-89 7a b1 c2 de 47 40 86   ./.%.qb..z...G@.
    0050 - 43 d9 a2 29 80 ef 95 ca-97 05 59 20 81 20 5b a5   C..)......Y . [.
    0060 - 6a d6 84 9a b5 a7 93 28-b8 97 fb bf 6b 3b 70 5a   j......(....k;pZ
    0070 - e4 5f a3 ad bf 4a 94 49-f9 5e 86 8b fe ab bc bd   ._...J.I.^......
    0080 - 33 17 42 7b 25 28 8f ff-54 e6 4b 50 f2 5d 82 f9   3.B{%(..T.KP.]..
    0090 - ef a7 38 cc a5 58 ef 1a-1a 98 ca a6 8a a6 ee be   ..8..X..........

    Start Time: 1517503410
    Timeout   : 300 (sec)
    Verify return code: 21 (unable to verify the first certificate)
---
DONE
```

**Comment by numismatika** [comment]
Thanks for the investigation, it at least makes sense now.

**Comment by J0WI** [comment]
So `bei.nuggad.net` should be excluded and documented. @numismatika please keep existing commits on your branch.

**Comment by yoshimo** [comment]
The problem is that other domains suffer from similiar problems. I doubt it is worth making a rule before nugg.ad has fixed their setup.

File: src/chrome/content/rules/Nugg.ad.xml [hunk 3]
```diff
@@ -1,31 +1,525 @@
...
-	<test url="http://si.nuggad.net/" />
-	<test url="http://yahoo-s.nuggad.net/" />
+<ruleset name="nugg.ad" >
+
+	<target host="nugg.ad" />
+	<target host="www.nugg.ad" />
+
+	<target host="io.nugg.ad" />
+	
+	<target host="yahoo-s.nuggad.net" />
```
Comment (injust) [inline src/chrome/content/rules/Nugg.ad.xml]: This host, `yahoo-s.nuggad.net`, appears again later in the ruleset. Please remove this occurrence.

**Comment by zoracon** [comment]
Closing due to conflicts in build for outdated PR. Leaving opportunity to reopen PR and assigning to myself for priority if it does.

**Pull request closed without merging**